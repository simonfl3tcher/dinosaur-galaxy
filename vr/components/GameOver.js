// React
import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
  VrButton,
} from 'react-vr';

export default class GameOver extends Component {
  render() {
    return (
      <View style={styles.gameStyle}>
        <Text style={styles.text}>Game Over!</Text>
        <Text style={styles.text}>You scored: {this.props.score}</Text>

        <VrButton style={styles.button} onClick={()=> this.props.startNewGame()}>
          <Text style={styles.buttonText}>Play Game!</Text>
        </VrButton>
      </View>
    )
  }
}

// Styles
const styles = StyleSheet.create({
  gameStyle: {
    transform: [
      { translate: [-2.25, 0, 0] }
    ]
  },
  button: {
    backgroundColor: '#fff',
    transform: [
      {translate: [0, 2, -5]}
    ]
  },
  text: {
    fontSize: 0.2,
    textAlign: 'center',
    color: '#fff',
    transform: [
      {translate: [0, 2, -5]}
    ]
  },
  buttonText: {
    fontSize: 0.2,
    textAlign: 'center',
    color: '#000'
  }
})
