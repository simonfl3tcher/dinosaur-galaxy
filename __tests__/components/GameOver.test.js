import 'react-native';
import 'react-vr';
import React from 'react';
import GameOver from '../../vr/components/GameOver.js';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<GameOver />).toJSON();
  expect(tree).toMatchSnapshot();
});
