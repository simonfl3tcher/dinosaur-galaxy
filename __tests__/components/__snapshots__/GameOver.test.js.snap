// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "transform": Array [
        Object {
          "translate": Array [
            -2.25,
            0,
            0,
          ],
        },
      ],
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    isOnLayer={false}
    style={
      Object {
        "color": "#fff",
        "fontSize": 0.2,
        "textAlign": "center",
        "transform": Array [
          Object {
            "translate": Array [
              0,
              2,
              -5,
            ],
          },
        ],
      }
    }
  >
    Game Over!
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    isOnLayer={false}
    style={
      Object {
        "color": "#fff",
        "fontSize": 0.2,
        "textAlign": "center",
        "transform": Array [
          Object {
            "translate": Array [
              0,
              2,
              -5,
            ],
          },
        ],
      }
    }
  >
    You scored: 
  </Text>
  <View
    disabled={false}
    ignoreLongClick={false}
    onClick={[Function]}
    onEnter={[Function]}
    onExit={[Function]}
    onInput={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "transform": Array [
          Object {
            "translate": Array [
              0,
              2,
              -5,
            ],
          },
        ],
      }
    }
    testID={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      isOnLayer={false}
      style={
        Object {
          "color": "#000",
          "fontSize": 0.2,
          "textAlign": "center",
        }
      }
    >
      Play Game!
    </Text>
  </View>
</View>
`;
